packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -112,6 +113,53 @@ export const walkDir = async (
     return fileSet
 }
 
+export const getAllProjectDependencyVersions = (): {[x: string]: any} => {
+    /**
+     * Returns the versions of all packages and their dependencies within the project.
+     *
+     * @returns An object representing the dependency tree
+     */
+    return JSON.parse(execSync(`npm ls --all --json`, {encoding: 'utf-8'}))
+}
+
+export const getLowestPackageVersion = (
+    packageName: string,
+    dependencyTree: {[x: string]: any}
+): string => {
+    /**
+     * Returns the lowest version of a package installed.
+     *
+     * @param packageName - The name of the package to get the lowest version for
+     * @param dependencyTree - The dependency tree including all package versions
+     * @returns The lowest version of the given package that is installed
+     */
+    let lowestVersion: string | null = null
+
+    function search(dependencyTree: {[x: string]: any}) {
+        for (const key in dependencyTree) {
+            if (key === packageName && dependencyTree[key].version) {
+                const version = dependencyTree[key].version
+                if (!lowestVersion || version < lowestVersion) {
```

Comment:
Versions aren't numbers, so you can't compare them with `<`. You'll need to parse the versions into major/minor/patch/prerelease and compare each part individually. This regex should help with that: `/^(\d+?)\.(\d+?)\.(\d+?)((?:-.+?)?)$/`

Reply:
Based on the tests, can we be confident this works as expected?

Reply:
`a < b` does not work for comparing semver strings. For example, JS says `1.10.0` < `1.9.0` and `1.2.3` < `1.2.3-beta`, but we know that the opposite is the case for semver comparisons.

Reply:
Ah I see! Thanks for explaining will fix this!